google/cloud/storage/internal/http_response.cc

```diff
@@ -186,11 +194,13 @@ Status AsStatus(HttpResponse const& http_response) {
   //     }
   //   }
   // See  https://cloud.google.com/apis/design/errors#http_mapping
-  auto error = json.value("error", nlohmann::json::object());
-  auto message = error.value("message", http_response.payload);
-  auto details = error.value("details", nlohmann::json::object());
-  auto error_info = MakeErrorInfo(details);
-  return Status(status_code, std::move(message), std::move(error_info));
+  if (!json.contains("error")) return default_error();
+  auto const& e = json["error"];
+  if (!e.is_object()) return default_error();
+  if (!e.contains("message") || !e.contains("details")) return default_error();
+  if (!e["message"].is_string()) return default_error();
```

Comment:
We could return a `Status(code, e.value("message", ""))` if `e.contains("message") && e["message"].is_string() && !e.contains("details")`, but it's also fine to return the whole payload as a message.

Reply:
Ack